tools/ddprobe.cpp

```diff
@@ -69,8 +71,128 @@ syncThreadDesktop() {
   CloseDesktop(hDesk);
 }
 
+/**
+ * @brief Checks whether a given frame is entirely dark by evaluating the RGB values of each pixel.
+ *
+ * This function determines if the provided frame is completely dark by analyzing the RGB values of every pixel.
+ * It iterates over all pixels in the frame and compares each pixel's RGB channels to a defined darkness threshold.
+ * If any pixel's RGB values exceed this threshold, the function concludes that the frame is not entirely dark and returns `false`.
+ * If all pixels are below the threshold, indicating a completely dark frame, the function returns `true`.
+ *
+ * @param mappedResource A reference to a `D3D11_MAPPED_SUBRESOURCE` structure containing the mapped subresource data of the frame to be analyzed.
+ * @param frameDesc A reference to a `D3D11_TEXTURE2D_DESC` structure describing the texture properties, including width and height.
+ * @param darknessThreshold A floating-point value representing the threshold above which a pixel's RGB values are considered non-dark. The value ranges from 0.0f to 1.0f, with a default value of 0.1f.
+ * @return bool Returns `true` if the frame is determined to be entirely dark, otherwise returns `false`.
+ */
+bool
+is_valid_frame(const D3D11_MAPPED_SUBRESOURCE &mappedResource, const D3D11_TEXTURE2D_DESC &frameDesc, float darknessThreshold = 0.1f) {
```

Comment:
I'm not sure of the validity of this function. It's perfectly valid (and common for dedicated gamestream servers) to have a perfectly black desktop with no icons or taskbar.

Reply:
Honestly, that does not matter because the code now defaults to the old behavior if it fails the frame test. The worst that could happen is that the bug will still exist, and that those users might need to disable the iGPU or add a single non-black pixel to their background.

I’d say that situation is extremely rare and unlikely to occur.